src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/TdsParserStaticMethods.cs

```diff
@@ -103,62 +111,92 @@ internal static byte[] ObfuscatePassword(string password)
                 s = (int)password[i];
                 bLo = (byte)(s & 0xff);
                 bHi = (byte)((s >> 8) & 0xff);
-                bEnc[i << 1] = (byte)((((bLo & 0x0f) << 4) | (bLo >> 4)) ^ 0xa5);
-                bEnc[(i << 1) + 1] = (byte)((((bHi & 0x0f) << 4) | (bHi >> 4)) ^ 0xa5);
+                bObfuscated[i << 1] = (byte)((((bLo & 0x0f) << 4) | (bLo >> 4)) ^ 0xa5);
+                bObfuscated[(i << 1) + 1] = (byte)((((bHi & 0x0f) << 4) | (bHi >> 4)) ^ 0xa5);
             }
-            return bEnc;
+            return bObfuscated;
         }
 
-        [ResourceExposure(ResourceScope.None)] // SxS: we use this method for TDS login only
-        [ResourceConsumption(ResourceScope.Process, ResourceScope.Process)]
-        static internal int GetCurrentProcessIdForTdsLoginOnly()
+        internal static byte[] ObfuscatePassword(byte[] password)
         {
-            return Common.SafeNativeMethods.GetCurrentProcessId();
+            byte bLo;
+            byte bHi;
+
+            for (int i = 0; i < password.Length; i++)
+            {
+                bLo = (byte)(password[i] & 0x0f);
+                bHi = (byte)(password[i] & 0xf0);
+                password[i] = (byte)(((bHi >> 4) | (bLo << 4)) ^ 0xa5);
+            }
+            return password;
         }
 
+        private const int NoProcessId = -1;
+        private static int s_currentProcessId = NoProcessId;
+        internal static int GetCurrentProcessIdForTdsLoginOnly()
```

Comment:
I chose the netcore implementation of GetCurrentProcessIdForTdsLoginOnly over the netfx one that relied on the deprecated method. I don't believe it will cause any noticeable behavior differences on netfx.